File: app/src/main/java/com/ndipatri/solarmonitor/container/HardwareModule.java
package com.ndipatri.solarmonitor.container;

import android.content.Context;

import com.ndipatri.solarmonitor.providers.panelScan.PanelScanProvider;

import javax.inject.Singleton;

import dagger.Provides;

@dagger.Module
public class HardwareModule {

    Context context;

    public HardwareModule(Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    PanelScanProvider providesBluetoothService(Context context) {
        return new PanelScanProvider(context);
    }
}

File: app/src/main/java/com/ndipatri/solarmonitor/container/HardwareModule.kt
package com.ndipatri.solarmonitor.container

import android.content.Context

import com.ndipatri.solarmonitor.providers.panelScan.PanelProvider

import javax.inject.Singleton

import dagger.Provides

@dagger.Module
class HardwareModule(internal var context: Context) {

    @Provides
    @Singleton
    internal fun providesBluetoothService(context: Context): PanelProvider {
        return PanelProvider(context)
    }
}
